File: HeadFirstDesignPatternsOC/DecoratorPattern/Mocha.m
//
//  Mocha.m
//  DecoratorPattern
//
//  Created by 陳峻琦 on 27/5/2022.
//

#import "Mocha.h"

@implementation Mocha

- (NSString *)description {
    return [NSString stringWithFormat:@"%@, Mocha", self.beverage];
}

- (double)cost {
    return 0.20 + [self.beverage cost];
}

@end

File: HeadFirstDesignPatternsJava/HeadFirst/src/DecoratorPattern/Mocha.java
package DecoratorPattern;

public class Mocha extends CondimentDecorator {
    Beverage beverage;

    public Mocha(Beverage beverage) {
        this.beverage = beverage;
    }

    @Override
    public String gerDescription() {
        return beverage.gerDescription() + "Mocha";
    }

    @Override
    public double cost() {
        return 0.20 + beverage.cost();
    }
    
}
